packages/strapi-plugin-upload/admin/src/containers/HomePage/reducer.js

```diff
@@ -2,34 +2,49 @@ import { fromJS } from 'immutable';
 
 const initialState = fromJS({
   data: [],
+  dataCount: 0,
   dataToDelete: [],
 });
 
 const reducer = (state, action) => {
   switch (action.type) {
+    case 'CLEAR_DATA_TO_DELETE':
+      return state.update('dataToDelete', () => fromJS([]));
     case 'GET_DATA_SUCCEEDED':
-      return state.update('data', () => fromJS(action.data));
+      return state
+        .update('data', () => fromJS(action.data))
+        .update('dataCount', () => action.count);
     case 'ON_CHANGE_DATA_TO_DELETE': {
       const { value, id } = action;
 
       if (value) {
+        const item = state.get('data').find(item => item.get('id') === id);
+
         return state.update('dataToDelete', dataToDelete => {
-          return dataToDelete.push(id);
+          return dataToDelete.push(item);
         });
       }
 
-      const index = state.get('dataToDelete').findIndex(item => item === id);
+      const index = state.get('dataToDelete').findIndex(item => item.get('id') === id);
 
       return state.removeIn(['dataToDelete', index]);
     }
     case 'TOGGLE_SELECT_ALL': {
-      const isSelected = state.get('data').size === state.get('dataToDelete').size;
+      const isSelected = state.get('data').every(item => state.get('dataToDelete').includes(item));
 
       if (isSelected) {
-        return state.update('dataToDelete', () => fromJS([]));
+        return state.update('dataToDelete', dataToDelete =>
+          dataToDelete.filter(item => !state.get('data').includes(item))
+        );
       }
 
-      return state.update('dataToDelete', () => state.get('data').map(item => item.get('id')));
+      return state.update('dataToDelete', dataToDelete => {
+        const newItems = state.get('data').filter(item => {
+          return !state.get('dataToDelete').includes(item);
```

Comment:
Same here

Reply:
I don't want `dataToDelete` to be an array of ids because we use an array of object to select the items in the modal list. We have to do it this way because of the pagination. We can select items though pagination but we also have to use all of them for the select tab display.

The list, which is the same component in the modal and the listView, now use an array of object and not an array of ids. I would prefer to be consistent and to let it as an array of objects to avoid multiple management cases inside a generic compo.